internal/app/uploader.go

```diff
@@ -1,33 +1,31 @@
 package app
 
 import (
+	"bufio"
 	"context"
 	"crypto/ecdsa"
+	"encoding/hex"
 	"fmt"
 	"io"
+	"log"
 	"os"
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/crypto"
 	"golang.org/x/crypto/sha3"
 )
 
-// BasinProviderUploader ...
-type BasinProviderUploader interface {
-	Upload(context.Context, string, string, uint64, io.Reader, *Signer, io.Writer, Timestamp) error
-}
-
 // BasinUploader contains logic of uploading Parquet files to Basin Provider.
 type BasinUploader struct {
 	namespace  string
 	relation   string
 	privateKey *ecdsa.PrivateKey
-	provider   BasinProviderUploader
+	provider   BasinProvider
```

Comment:
these are private fields but do we want to keep using the old names here or it doesn't matter?

Reply:
yeah, makes sense to change to `VaultsProvider` and `VaultsUploader`, we can iterate on that on the next PRs